edge-agent/docker/linux/amd64/Dockerfile

```diff
@@ -1,7 +1,7 @@
-ARG base_tag=2.1.17-alpine3.11
+ARG base_tag=3.1.4-alpine3.11
 FROM azureiotedge/azureiotedge-runtime-base:1.2-linux-amd64 as builder
```

Comment:
Do we need to update runtime base image?

Reply:
I was thinking no, because this base tag describing dotnet version is not used there. We could update alpine to 3.11 from 3.10 but I think this is also fine since it is used only for building.